calcom/cal.com
# fix: multiple duration getSchedule calls [CAL-2336]

## What does this PR do?

Due to the nature of multiple duration and recent changes that aimed to call as fast as possible the `getSchedule` result to populate the available slots, that call was made multiple times, first with an undefined duration and then the selected one.

Now we preemptively get the duration from query param if present, check if it's a possibility within the given event type options and return a valid duration if the event type has multiple duration configuration, either the query parameter value or the default duration if not valid. That duration is then passed down to the `getSchedule` call to effectively gain the performance optimization that was expected.

Fixes #10707 

## Type of change




## How should this be tested?

Go to `/pro/multiple-duration` event type and check in Dev Tools Network tab that `getSchedule` is called once when loading the page. Before this change it was called twice.

## Mandatory Tasks
